# HG changeset patch
# User Mike Miller <mtmiller@ieee.org>
# Date 1424752117 18000
#      Mon Feb 23 23:28:37 2015 -0500
# Node ID 07b970d6cdfe3167da74109871eb5629cf387ddb
# Parent  4006a7377a190ef6b975574f5c2038b497a3d294
Only install config.log in $octetc if user requested (bug #43087)

* configure.ac: Add a --enable-install-build-logs configure option.
Set AMCOND_INSTALL_BUILD_LOGS.
* Makefile.am (octetc_DATA): Use AMCOND_INSTALL_BUILD_LOGS to include
config.log in the list conditionally.

diff --git a/Makefile.am b/Makefile.am
--- a/Makefile.am
+++ b/Makefile.am
@@ -168,18 +168,21 @@ define changelog-from-hg-log
 endef
 
 ChangeLog:
 	$(AM_V_GEN)$(changelog-from-hg-log)
 .PHONY: ChangeLog
 
 octetc_DATA = \
   CITATION \
-  NEWS \
-  config.log
+  NEWS
+
+if AMCOND_INSTALL_BUILD_LOGS
+octetc_DATA += config.log
+endif
 
 doxyhtml:
 	$(MAKE) -C doc/doxyhtml doxyhtml
 .PHONY: doxyhtml
 
 DIRS_TO_MAKE = \
   $(localfcnfiledir) \
   $(localapifcnfiledir) \
diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -2960,16 +2960,25 @@ fi
 AM_CONDITIONAL([AMCOND_BUILD_GUI], [test $build_gui = yes])
 AM_CONDITIONAL([AMCOND_HAVE_QSCINTILLA],
                [test "$octave_cv_lib_qscintilla" = yes])
 AM_CONDITIONAL([WIN32_TERMINAL], [test $win32_terminal = yes])
 AC_SUBST(QT_CPPFLAGS)
 AC_SUBST(QT_LDFLAGS)
 AC_SUBST(QT_LIBS)
 
+### Decide whether or not to install build logs with Octave.
+
+install_build_logs=no
+AC_ARG_ENABLE([install-build-logs],
+  [AS_HELP_STRING([--enable-install-build-logs],
+    [install build logs (e.g. config.log) with Octave])],
+  [if test "$enableval" = yes; then install_build_logs=yes; fi])
+AM_CONDITIONAL([AMCOND_INSTALL_BUILD_LOGS], [test $install_build_logs = yes])
+
 CFLAGS="$original_octave_configure_CFLAGS"
 CXXFLAGS="$original_octave_configure_CXXFLAGS"
 
 AH_BOTTOM([#include "oct-conf-post.h"])
 
 ### Make all AC_DEFINES available to testif feature of test.m function.
 ### This must reside at the bottom of configure.ac after all AC_DEFINES
 ### have been made.
